src/qobj/quantum_object.jl

```diff
@@ -294,7 +190,7 @@ function Base.show(
     return show(io, MIME("text/plain"), op_data)
 end
 
-function Base.show(io::IO, QO::QuantumObject{<:AbstractArray{T},OpType}) where {T,OpType<:OperatorQuantumObject}
+function Base.show(io::IO, QO::AbstractQuantumObject)
```

Comment:
Regarding to `show`
This is the output of `QobjEvo` from python qutip:
```
QobjEvo: dims = [[2], [2]], shape = (2, 2), type = oper, superrep = None, isconstant = False, num_elements = 3
```

Should we separate the method of `show` for `Qobj` and `QobjEvo` ?
That is, rebase `Qobj`'s `show` definition as
```julia
Base.show(io::IO, QO::QuantumObject{<:AbstractArray{T},OpType}) where {T,OpType<:OperatorQuantumObject}
```
and maybe print something like the following for `QobjEvo`:
```
Quantum Object Evolution:   type=Operator,   dims=[2, 2],   size=(2, 2),   ishermitian=true,   isconstant=false, 
  num_elements = 3
# data print of SciMLOperator
```
For the `num_elements`, I only know that we can access `ops` from `AddedOperator` and get the `length` of it, namely
```julia
H::AddedOperator # or TensorProductOperator
length(H.ops)
```

But I'm not sure how to get it from pure `MatrixOperator`.
Or maybe we need to define a new function (maybe call it `_get_QobjEvo_num_element`) and multiple dispatch for different `AbstractSciMLOperator`.

Reply:
For the moment I would leave it as it is. Maybe in a next PR we can do this.